src/server.cpp

```diff
@@ -1054,10 +1056,15 @@ void CServer::DecodeReceiveData ( const int iChanCnt, const int iNumClients )
                     emit ClientDisconnected ( iCurChanID ); // TODO do this outside the mutex lock?
                 }
 
+                FreeChannel ( iCurChanID ); // note that the channel is now not in use
+
                 // note that no mutex is needed for this shared resource since it is not a
                 // read-modify-write operation but an atomic write and also each thread can
                 // only set it to true and never to false
                 bChannelIsNowDisconnected = true;
+
+                // since the channel is no longer in use, we should return
+                return;
```

Comment:
Without leaving early, it will continue to dereference `DoubleFrameSizeConvBufIn[iCurChanID]`, which I felt wasn't necessarily correct, and probably unnecessary anyway. The function is only executing on behalf of a single channel, that had now been disconnected, so I didn't think it needed to continue.

Reply:
OK.  I've always felt there was some odd code flow in it... and been wary of interfering :).